dt_model/model/abstract_model.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from dt_model.symbols.context_variable import ContextVariable
+from dt_model.symbols.presence_variable import PresenceVariable
+from dt_model.symbols.index import Index
+from dt_model.symbols.constraint import Constraint
+
+
+class AbstractModel:
```

Comment:
For a future commit or pull request, we should considering refactoring this class to be an immutable data class using `@dataclass(frozen=True)`, which provides strong immutability guarantees and, in my opinion, seems to map quite nicely with the semantics we are imposing around here.